model.py

```diff
@@ -7953,8 +7953,8 @@ class CustomList(Base):
     id = Column(Integer, primary_key=True)
     primary_language = Column(Unicode, index=True)
     data_source_id = Column(Integer, ForeignKey('datasources.id'), index=True)
-    foreign_identifier = Column(Unicode, index=True)
-    name = Column(Unicode)
+    foreign_identifier = Column(Unicode, index=True, unique=True)
+    name = Column(Unicode, index=True, unique=True)
```

Comment:
Can (data_source_id, foreign_identifier) and (data_source_id, name) each have a unique constraint? I can see different data sources each providing a generic list name like 'Best Sellers'.